processor/schemaprocessor/internal/schema/manager.go

```diff
@@ -0,0 +1,209 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+
+	schemaencoder "go.opentelemetry.io/otel/schema/v1.0"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+const (
+	DefaultConcurrentFetches = 10
+)
+
+var (
+	errNilValueProvided = errors.New("nil value provided")
+)
+
+// Manager is responsible for ensuring that schemas are kept up to date
+// with the most recent version that are requested.
+type Manager interface {
```

Comment:
Nit: can we give this a better name? Manager doesn't sound specific enough. Perhaps `Fetcher` meaning that it fetches schemas?

Reply:
I went with Manager because it stores and manages updates to translations, fetching them and eventually where to find those (via an collector extension or over the network).

Though, having been working on this, my idea is has slightly changed.

My intention is that I am renaming this package `schema --> translation` since it feels more fitting on what is trying to achive, and the potential renaming this to `Storer` since at runtime is stores the collection of schemaURL (With their translations) and will provide async updates as they are needed.